lib/events/kafka_listener.rb

```diff
@@ -26,21 +26,42 @@ def subscribe
       Rails.logger.error(["Something is wrong with Kafka client: ", e.message, *e.backtrace].join($RS))
       retry
     end
-  
+
     private
-  
+
     def raw_process(event)
       Rails.logger.info("Kafka message #{event.message} received with payload: #{event.payload}")
-  
-      ActiveRecord::Base.connection_pool.with_connection do
-        process_event(event)
+
+      insights_headers = event.headers.slice('x-rh-identity', 'x-rh-insights-request-id')
+      if insights_headers.empty?
+        header_hash = JSON.parse(event.payload)
+        insights_headers['x-rh-insights-request-id'] = header_hash["request_id"]
+        insights_headers['x-rh-identity'] = header_hash["b64_identity"]
+      end
+
+      unless insights_headers['x-rh-identity'] && insights_headers['x-rh-insights-request-id']
+        Rails.logger.error("Message skipped because of missing required headers")
+        return
+      end
+
+      Insights::API::Common::Request.with_request(:headers => insights_headers, :original_url => nil) do |req|
+        tenant = Tenant.first_or_create(:external_tenant => req.tenant)
```

Comment:
@bzwei This is needed because we are creating tenant and source records based on messages coming from Sources API, without this change we would never be able to create new Tenant and the Source create will fail. The earlier code was assuming that the Tenant always exists, in the Catalog Inventory API that's not true we will never have Tenant and Sources till we start receiving messages from Sources API on the Kafka Topic **platform.sources.event-stream**